src/tactic/lean_core_docs.lean

```diff
@@ -0,0 +1,647 @@
+
+add_tactic_doc
+{ name       := "apply_opt_param",
+  category   := doc_category.tactic,
+  decl_names := [`tactic.interactive.apply_opt_param],
+  tags       := ["core"] }
+
+add_tactic_doc
+{ name       := "apply_with",
+  category   := doc_category.tactic,
+  decl_names := [`tactic.interactive.apply_with],
+  tags       := ["core"] }
+
+add_tactic_doc
+{ name       := "assume",
+  category   := doc_category.tactic,
+  decl_names := [`tactic.interactive.assume],
+  tags       := ["core"] }
+
+add_tactic_doc
+{ name       := "assumption",
+  category   := doc_category.tactic,
+  decl_names := [`tactic.interactive.assumption],
+  tags       := ["core", "basic"] }
+
+add_tactic_doc
+{ name       := "assumption'",
+  category   := doc_category.tactic,
+  decl_names := [`tactic.interactive.assumption'],
+  tags       := ["core", "goal management"] }
+
+add_tactic_doc
+{ name       := "async",
+  category   := doc_category.tactic,
+  decl_names := [`tactic.interactive.async],
+  tags       := ["core"] }
+
+/--
+`by_cases (h :)? p` splits the main goal into two cases, assuming `h : p` in the first branch, and `h : ¬ p` in the second branch.
+
+This tactic requires that `p` is decidable. To ensure that all propositions are decidable via classical reasoning, use
+`open_locale classical` (or `local attribute [instance, priority 10] classical.prop_decidable` if you are not using mathlib).
+-/
+add_tactic_doc
+{ name       := "by_cases",
+  category   := doc_category.tactic,
+  decl_names := [`tactic.interactive.by_cases],
+  tags       := ["core", "basic"] }
+
+/--
+If the target of the main goal is a proposition `p`, `by_contra h` reduces the goal to proving `false` using the additional hypothesis `h : ¬ p`. If `h` is omitted, a name is generated automatically.
+
+This tactic requires that `p` is decidable. To ensure that all propositions are decidable via classical reasoning, use
+ `open_locale classical` (or `local attribute [instance, priority 10] classical.prop_decidable` if you are not using mathlib).
+-/
+add_tactic_doc
+{ name       := "by_contra",
+  category   := doc_category.tactic,
+  decl_names := [`tactic.interactive.by_contra, `tactic.interactive.by_contradiction],
+  tags       := ["core"] }
+
+add_tactic_doc
+{ name       := "case",
+  category   := doc_category.tactic,
+  decl_names := [`tactic.interactive.case],
+  tags       := ["core", "goal management"] }
+
+add_tactic_doc
+{ name       := "cases",
+  category   := doc_category.tactic,
+  decl_names := [`tactic.interactive.cases],
+  tags       := ["core", "basic"] }
+
+/--
+`cases_matching p` applies the `cases` tactic to a hypothesis `h : type` if `type` matches the pattern `p`.
+`cases_matching [p_1, ..., p_n]` applies the `cases` tactic to a hypothesis `h : type` if `type` matches one of the given patterns.
+`cases_matching* p` more efficient and compact version of `focus1 { repeat { cases_matching p } }`. It is more efficient because the pattern is compiled once.
```

Comment:
```suggestion
`cases_matching* p` is a more efficient and compact version of `focus1 { repeat { cases_matching p } }`. It is more efficient because the pattern is compiled once.
```